### DISCUSSION TITLE
Appium with flutter

### DISCUSSION THREAD

**Discussion by CorentinSailly62** [appium/appium#16122] [General]
Is appium able to analyse the keys of the elements of an application developed in flutter? Or does appium not have the functionality for this?

**Comment by jlipps** [comment]
What are the "keys of the elements"? Have you tried the Appium Flutter driver?

**Reply by CorentinSailly62** [reply]
a keys element like
 this "   child: Text(
        'Go to next route',
        key: Key('next_route_key'),
      ),"


I would like to know if there is an appium desktop that allows to analyse the content and get the flutter keys in the views? :)

**Reply by CorentinSailly62** [reply]
is there a spyware appium that allows to do that

**Reply by jlipps** [reply]
I don't think so, but you could ask at the flutter driver repo (we don't maintain the flutter driver).

**Reply by KazuCocoa** [reply]
You can call `getRenderTree` execute command to get the flutter's view hierarchy, but it is not XML format. Appium Desktop only can print the response as a response popup view with Flutter driver.
https://github.com/appium-userland/appium-flutter-driver#commands
